cpp/include/rmm/mr/device/cuda_async_memory_resource.hpp

```diff
@@ -48,28 +48,37 @@ class cuda_async_memory_resource final : public device_memory_resource {
   /**
    * @brief Flags for specifying memory allocation handle types.
    *
-   * @note These values are exact copies from `cudaMemAllocationHandleType`. We need to
-   * define our own enum here because the earliest CUDA runtime version that supports asynchronous
-   * memory pools (CUDA 11.2) did not support these flags, so we need a placeholder that can be
-   * used consistently in the constructor of `cuda_async_memory_resource` with all versions of
-   * CUDA >= 11.2. See the `cudaMemAllocationHandleType` docs at
+   * @note These values are exact copies from `cudaMemAllocationHandleType`. We need a placeholder
+   * that can be used consistently in the constructor of `cuda_async_memory_resource` with all
+   * supported versions of CUDA. See the `cudaMemAllocationHandleType` docs at
    * https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html and ensure the enum
    * values are kept in sync with the CUDA documentation.
+   *
+   * @note cudaMemHandleTypeFabric can be used instead of 0x8 once we require
+   * CUDA 12.4+.
```

Comment:
Ah, that explains why we updated all the others to use enum values, except for `0x8`.